elements/rh-cta/rh-cta.ts

```diff
@@ -0,0 +1,148 @@
+import type { ColorPalette, ColorTheme } from '../../lib/context/color.js';
+
+import { LitElement, html } from 'lit';
+import { customElement, property } from 'lit/decorators.js';
+
+import { Logger } from '@patternfly/pfe-core/controllers/logger.js';
+
+import { DirController } from '../../lib/DirController.js';
+import { colorContextConsumer, colorContextProvider } from '../../lib/context/color.js';
+import { classMap } from 'lit/directives/class-map.js';
+
+import '@patternfly/pfe-icon';
+
+import style from './rh-cta.css';
+
+export interface CtaData {
+  href?: string;
+  text?: string;
+  title?: string;
+  color?: string;
+  type?: string;
+}
+
+const supportedTags = ['a', 'button']; // add input later
+function isSupportedContent(el: Element|null): el is HTMLAnchorElement|HTMLButtonElement {
+  return !!el && supportedTags.includes(el.localName);
+}
+
+const CONTENT = new WeakMap<Element, boolean>();
+function contentInitialized(el: Element|null): boolean {
+  return !!el && !!CONTENT.get(el);
+}
+
+function isButton(element: Element): element is HTMLButtonElement {
+  return element.tagName.toLowerCase() === 'button';
+}
+
+/**
+ * Call to action stands out from regular hypertext links, and is used for linking users to webpages.
+ *
+ * @summary Directs a user to other pages or displays extra content
+ *
+ * @slot - We expect an anchor tag, `<a>` with an `href`, to be the first child inside `rh-cta` element. Less preferred but allowed for specific use-cases include: `<button>` (note however that the `button` tag is not supported for the default CTA styles).
+ *
+ * @fires {CtaSelectEvent} select - This event is fired when a link is clicked and serves as a way to capture click events if necessary.
+ *
```

Comment:
```suggestion
```

We should remove this event as discussed here: https://github.com/RedHat-UX/red-hat-design-system/issues/333#issuecomment-1209653112

Reply:
removed